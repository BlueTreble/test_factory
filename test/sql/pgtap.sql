\set ECHO none
\i test/helpers/setup.sql

SET search_path = tap;
-- IF YOU GET A "schema tf does not exist" error here then the dependency is missing!
SELECT throws_ok(
  $$CREATE EXTENSION test_factory_pgtap$$
  , '42704'
  , 'required extension "test_factory" is not installed'
  , 'Ensure test_factory is a dependency of test_factory_pgtap'
);

\set extension_name test_factory
\i test/helpers/create_extension.sql
DROP TABLE pre_install_role;
DROP TABLE post_install_role;
\set extension_name test_factory_pgtap
\i test/helpers/create_extension.sql

-- NOTE: This runs some tests itself. It also changes search_path
\i test/helpers/create.sql

-- tf.tap already returns tap output
SELECT tf.tap( 'invoice' );
SELECT tf.tap( 'invoice', 'base' );
SELECT throws_ok(
  $$SELECT tf.tap( '"non-existent table"' )$$
  , '42P01'
  , 'relation "non-existent table" does not exist'
  , 'Ensure we get sane error for a non-existent table'
);

ROLLBACK;

-- vi: expandtab ts=2 sw=2
