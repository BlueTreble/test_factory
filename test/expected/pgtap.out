\set ECHO none
ok 1 - Ensure test_factory is a dependency of test_factory_pgtap
Creating extension test_factory
Creating extension test_factory_pgtap
ok 2 - Register test customers
ok 3 - Create function customer__add
ok 4 - Register test invoices
ok 5 - Ensure original_role temp table was dropped
ok 6 - Ensure role is put back after install
ok 7 - Security definer function _tf.schema__getsert has search_path=pg_catalog
ok 8 - Security definer function _tf.test_factory__get has search_path=pg_catalog
ok 9 - Security definer function _tf.test_factory__set has search_path=pg_catalog
ok 10 - Security definer function _tf.table_create has search_path=pg_catalog
ok 11 - Security definer function _tf.get has search_path=pg_catalog
ok 12 - Get test data set "base" for table invoice
ok 13 - Get test data set "base" for table invoice
ok 14 - Ensure we get sane error for a non-existent table
